src/Bicep.Cli/Commands/BuildCommand.cs

```diff
@@ -66,11 +66,13 @@ public async Task<DiagnosticSummary> CompileMultiple(BuildArguments args)
     {
         var hasErrors = false;
 
+        var (inputDir, _) = CommandHelper.SplitFilePatternOnWildcard(args.FilePattern!, Path.DirectorySeparatorChar);
```

Comment:
The interaction with `--pattern` is a bit subtle - essentially if we have a path like:
```
--pattern /path/to/**/*.bicep
```

Then the part before the first wildcard is considered the "base path", which will be replaced with the output dir.

I'm wondering if it's worth making this behavior more explicit if used with the `--outdir` argument. Some possible options:
* Go with the implementation proposed here (I might be overthinking it!)
* Introduce an argument to explicitly specify the base dir (e.g.`--pattern-basedir`) which must be used with `--outdir`. Require relative paths for the `--pattern` argument if it's used.

@jeskew / @shenglol - do you have opinions on this?

Reply:
Just some additional thoughts here:

the two params cover all cases, as if you want to keep the prefix in the output path, you can just repeat it, ie `--outdir /path/to/`.

Imo adding another param makes the UX more confusing as it doubles the possible combinations of alloweds params.

Reply:
The interaction seems intuitive to me as well. I think the MSBuild Copy task does the same thing.

Reply:
Thanks for the discussion! I'm good with this.